registry/storage/linkedblobstore.go

```diff
@@ -74,6 +75,20 @@ func (lbs *linkedBlobStore) Put(ctx context.Context, mediaType string, p []byte)
 		return distribution.Descriptor{}, err
 	}
 
+	// Remove a tombstone if one exists for this blob.  Continue to write
+	// the blob as it may have been removed by the GC sweep process.  If
+	// deletes are not enabled, fall through
+	dgst, err := digest.FromBytes(p)
```

Comment:
I think this code can be moved to `linkedBlobStatter` and replaced with a call to `SetDescriptor`, which would clear the tombstone.